plugins/polio/models/performance_thresholds.py

```diff
@@ -0,0 +1,133 @@
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    # Managers
+    objects = DefaultSoftDeletableManager.from_queryset(PerformanceThresholdQuerySet)()
+    objects_only_deleted = OnlyDeletedSoftDeletableManager.from_queryset(PerformanceThresholdQuerySet)()
+    objects_include_deleted = IncludeDeletedSoftDeletableManager.from_queryset(PerformanceThresholdQuerySet)()
+
+    class Meta:
+        verbose_name = _("Performance Threshold")
+        verbose_name_plural = _("Performance Thresholds")
+        ordering = ["indicator"]
+        unique_together = ["account", "indicator"]
+
+        indexes = [
+            models.Index(fields=["account"]),
+            models.Index(fields=["indicator"]),
+        ]
+
+    def __str__(self):
+        return f"{self.account.name}-{self.indicator}"
+
+    # make method static for easier testing
+    @staticmethod
+    def is_json_logic_rule(data):
+        if not data:
+            return False
+        keys = data.keys()
+        values = list(data.values())
+        operators = ["==", ">", "<", ">=", "<="]
+        unsupported_keys = [key for key in keys if key not in operators]
+        all_keys_valid = not unsupported_keys
+        correct_array_format = len([value for value in values if (isinstance(value, list) and len(value) == 2)]) == len(
+            keys
+        )
+        correct_reference_format = False
+        correct_argument_format = False
+        if correct_array_format:
+            correct_reference_format = len([value for value in values if (value[0] == {"var": "value"})]) == len(keys)
+
+            correct_argument_format = len(
+                [value for value in values if (value[1] == {"var": "average"} or isinstance(value[1], int))]
+            ) == len(keys)
+
+        if all_keys_valid and correct_array_format and correct_reference_format and correct_argument_format:
+            return True
+
+        return False
+
+    # make method static for easier testing
+    @staticmethod
+    def is_json_logic_expression(data):
+        if not data:
+            return False
+        keys = data.keys()
+
+        if "and" not in keys and "or" not in keys:
+            return PerformanceThresholds.is_json_logic_rule(data)
+
+        values = data.values()
+        found_invalid_data = False
+        for entry in values:
+            if not isinstance(entry, list):
+                return False
+            entry_keys = sum([list(rule.keys()) for rule in entry], [])
+            # this model does not support nested operators
+            if "and" in entry_keys or "or" in entry_keys:
+                found_invalid_data = True
+            else:
+                found_invalid_data = (
+                    len([rule for rule in entry if not PerformanceThresholds.is_json_logic_rule(rule)]) > 0
+                )
+            if found_invalid_data:
+                break
+        return not found_invalid_data
```

Comment:
Since you don't need to return which part is invalid, I'd suggest simplifying this part:

```suggestion
        if not data:
            return False
        for key, value in data.items():
            if key not in ["and", "or"]:
                # Single rule
                return PerformanceThresholds.is_json_logic_rule(data)
            if not isinstance(value, list):
                return False
            for entry in value:
                entry_keys = entry.keys()
                if "and" in entry_keys or "or" in entry_keys:
                    # this model does not support nested operators
                    return False
                if not PerformanceThresholds.is_json_logic_rule(entry):
                    return False
        return True
```

Reply:
Same, I prefer to return `False` by default when validating, especially with complex logic